File: pset6/caesar/caesar.py
import sys
import string

if len(sys.argv) != 2:
    print("Usage: python caesar.py <shift>")
    exit(1)

try:
    shift = int(sys.argv[1])
except:
    print("Shift needs to be a positive integer.")
    exit(1)

plain = input("plaintext: ")

cipher = ""

for c in plain:
    if c.isupper():
        cipher += string.ascii_uppercase[(string.ascii_uppercase.index(c) + shift) % 26]
    elif c.islower():
        cipher += string.ascii_lowercase[(string.ascii_lowercase.index(c) + shift) % 26]
    else:
        cipher += c

print("ciphertext: " + cipher)

File: pset2/caesar/caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
    int key = atoi(argv[1]) % 26;
    string plain = get_string("plaintext: ");
    printf("ciphertext: ");
    for (int i = 0; i < strlen(plain); i++)
    {
        if (isalpha(plain[i]))
        {
            int letter_c = (int) plain[i] + key;
            if (letter_c > 122)
            {
                letter_c -= 26;
            }
            printf("%c", letter_c);
        }
        else
        {
            printf("%c", plain[i]);
        }
    }
    printf("\n");
    return 0;
}